app/main.py

```diff
@@ -1,8 +1,59 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+    def __init__(self, name, age):
+        self.name = name
+        self.age = age
+        self.wife = None
+        self.husband = None
+
+        Person.people[name] = self
+
+
+    def set_spouse(self, spouse):
+        if self.age < spouse.age:
+            self.wife = spouse
+            spouse.husband = self
+        else:
+            self.husband = spouse
+            spouse.wife = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_instances = []
+    for person in people:
+        new_person = Person(person["name"], person["age"])
+
+    for person in people:
+        new_person = Person.people[person["name"]]
+
+        if "wife" in person and person["wife"]:
+            spouse = Person.people[person["wife"]]
+            new_person.set_spouse(spouse)
+        elif "husband" in person and person["husband"]:
+            spouse = Person.people[person["husband"]]
+            new_person.set_spouse(spouse)
+    return list(Person.people.values())
```

Comment:
The function returns `list(Person.people.values())`, which might not reflect the intended behavior if `Person.people` is used for other purposes. Consider returning `person_instances` instead, after ensuring it contains all created `Person` instances.